pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -526,11 +527,10 @@ func hashRouteList(r []*route.Route) uint64 {
 		default:
 			binary.LittleEndian.PutUint64(b, uint64(u))
 		}
-		hash.Write(b)
+		h.Write(b)
 	}
 	var tmp [md5.Size]byte
```

Comment:
`md5.Size` is suspicious since its not md5 anymore?

BTW, we never pre-alloc the bytes so we could make the functions not take an arg

Reply:
I noticed it, and I don't think the tmp here has any optimization effect, but I'm not sure if removing it will cause problems. [Here](https://groups.google.com/g/golang-nuts/c/2PwoTdJz_JY) are some discussions about pre-alloc. Can I change it to `h.ToUint64()`?

Reply:
It only optimizes if we reuse it. Since we don't , it provides no value. I would just make it `ToUint64()` for now and if we have use cases where we could pre-alloc it we can add a new interface/change the existing one

Reply:
Agreed, I will submit later.

Reply:
Commit pushed